esmvalcore/cmor/_fixes/native6/era5.py

```diff
@@ -390,10 +515,25 @@ def _fix_coordinates(self, cube):
                 and len(coord.core_points()) > 1
                 and coord_def.must_have_bounds == "yes"
             ):
-                coord.guess_bounds()
+                # Do not guess bounds for lat and lon on unstructured grids
+                if not (
+                    coord.name() in ("latitude", "longitude")
+                    and has_unstructured_grid(cube)
+                ):
+                    coord.guess_bounds()
 
         self._fix_monthly_time_coord(cube)
 
+        # Fix coordinate increasing direction
+        if cube.coords("latitude") and not has_unstructured_grid(cube):
+            lat = cube.coord("latitude")
+            if lat.points[0] > lat.points[-1]:
```

Comment:
Why is this additional check needed? It does not seem to be applied to grib data because that is on an unstructured grid and for NetCDF data it should just be applied

Reply:
Klaus once used a smart word to describe that our fixes should only fix wrong data, but do nothing for correct data (I can't remember what it was). This is what I'm trying to do here. If someone applies this fix to data with a correctly ordered lat, this fix should not reverse it.

Reply:
Okay, ChatGPT tells me it's "idempotent" 😁

Reply:
I can see that is a nice feature, but overly defensive code increases the maintenance burden. E.g. what if someone tries to use the data of only a single latitude point? (as we've had with the time dimension #2511)

Reply:
Anyway, I'm fine with leaving this as is.

Reply:
> E.g. what if someone tries to use the data of only a single latitude point? (as we've had with the time dimension #2511)

Nothing would happen since using the index `-1` on a list with one element is totally fine, and the if statement would simply evaluate to `False`.

Reply:
not quite: idempotence is when you apply an operator, then no matter however many times you apply the same operator, the result doesn't change anymore, past the first result